lib/moonshot/stack.rb

```diff
@@ -176,6 +188,25 @@ def add_parameter_overrides(hash)
 
     private
 
+    def asgs
+      resources_of_type('AWS::AutoScaling::AutoScalingGroup')
+    end
+
+    def instance_ip(instance_id)
+      Aws::EC2::Client.new.describe_instances(instance_ids: [instance_id])
```

Comment:
This has the potential to raise a variety of non-intuitive errors. For example:
- What if EC2 and AutoScaling don't agree on the state of the instance and Reservations[].Instances[] comes back empty?
- What if the instance is still booting and doesn't have a public IP address?

I think as a first pass we could just rescue StandardError (i.e. naked `rescue`) and re-raise a RuntimeError that says "Failed to determine Instance IP."